### PR TITLE
Refactor authflow nodes as intent for bot protection subflow

### PR DESCRIPTION
**Pull request by pkong-ds**
## What's in this PR?
1. Refactored existing nodes in below auth flow branches as intents
2. Added some e2e tests for auth flow branches affected (ref DEV-1484)

## Why this PR?
To accommodate upcoming changes in bot protection.

Since a simple node cannot spawn new sub flow, these branches nodes need to be `intent` to spawn `IntentBotProtection`

## Questions [ALL resolved]

~@louischan-oursky Can help with below when free?~

1. [[RESOLVED in below comment](https://github.com/authgear/authgear-server/pull/4395#issuecomment-2188679171)] Could not found `intent_PROMOTE_flow_step_create_authenticator`, is the flow reusing `intent_signup_...`?
3. [RESOLVED with newman] now `make -C e2e run` always timeout after 10 min after running locally, tried below to no avail
   - `make -C e2e teardown` first
   - remove docker image
   Suspect this timeout is related to changes in this PR, but the timeout seem no logs to trace. How should one debug this? 
4. [[RESOLVED in tung's comment](https://github.com/authgear/authgear-server/commit/f8bfb120833cf81b72996307e0f272713ae8f267#r143503074)]Not sure about the refactoring of `node_create_authenticator_totp` in f8bfb120833cf81b72996307e0f272713ae8f267 , please advise if any 🙏

### CODE REVIEW

**Comment by tung2744** [comment]
1. Yes. You can read pkg/lib/authenticationflow/declarative/intent_promote_flow_steps.go

File: pkg/lib/authenticationflow/declarative/intent_create_authenticator_totp.go [hunk 1]
```diff
@@ -82,19 +82,19 @@ func NewNodeCreateAuthenticatorTOTP(deps *authflow.Dependencies, n *NodeCreateAu
...
 
-func (*NodeCreateAuthenticatorTOTP) Kind() string {
-	return "NodeCreateAuthenticatorTOTP"
+func (*IntentCreateAuthenticatorTOTP) Kind() string {
+	return "IntentCreateAuthenticatorTOTP"
 }
 
-func (*NodeCreateAuthenticatorTOTP) Milestone() {}
-func (*NodeCreateAuthenticatorTOTP) MilestoneFlowCreateAuthenticator(flows authflow.Flows) (MilestoneDoCreateAuthenticator, authflow.Flows, bool) {
+func (*IntentCreateAuthenticatorTOTP) Milestone() {}
```
Comment (tung2744) [inline pkg/lib/authenticationflow/declarative/intent_create_authenticator_totp.go:89]: https://github.com/authgear/authgear-server/commit/f8bfb120833cf81b72996307e0f272713ae8f267#r143503074

**Comment by pkong-ds** [comment]
Changed 14 nodes into intents

```text
intent_create_authenticator_password
intent_create_authenticator_totp
intent_create_identity_login_id
intent_lookup_identity_login_id
intent_lookup_identity_passkey
intent_promote_identity_login_id
intent_skip_creation_by_existing_identity
intent_use_account_recovery_identity
intent_use_authenticator_passkey
intent_use_authenticator_password
intent_use_authenticator_totp
intent_use_identity_login_id
intent_use_identity_passkey
intent_use_recovery_code
```

**Comment by pkong-ds** [comment]
Added below e2e tests, reauth still WIP; Feel free to review first

- Signup flows
  - create_authenticator
    - totp
- Login Flows
  - authenticate
    - totp

- Signup-Login flows

  - For non-existing, switch to Signup

  - For existing, switch to Login

- Account-Recovery flows (Change password)

  - entire flow + login

**Comment by pkong-ds** [comment]
Since it is agreed that below will not have e2e tests (see DEV-1484), please find below a screenshot video on these 2 features working locally

## `signin > create_authentifcator > passkey`
https://github.com/authgear/authgear-server/assets/74223769/f4b1c2e6-0bf2-4641-bde9-b5893fb1f5b2
## `login > authenticate > passkey` 
https://github.com/authgear/authgear-server/assets/74223769/aea4ddd5-a937-4af3-82f1-8a0b9257ff6c

**Comment by pkong-ds** [comment]
@louischan-oursky ready for review 🙏 local ci & e2e pass

File: e2e/tests/login/totp/totp_user.sql [hunk 2]
```diff
@@ -0,0 +1,301 @@
+INSERT INTO
```
[the remark this answers is not in the document]
Comment (pkong-ds) [reply e2e/tests/login/totp/totp_user.sql:1]: @louischan-oursky Thanks, will do it for login_totp

For TOTP recovery code, is there a way to import user with TOTP recovery code?

Checked seems only secret is available https://github.com/authgear/authgear-server/blob/main/pkg/lib/userimport/model.go#L55

[the remark this answers is not in the document]
Comment (pkong-ds) [reply e2e/tests/login/totp/totp_user.sql]: Added [541257b](https://github.com/authgear/authgear-server/pull/4395/commits/541257b55977c32630ccdefa8fc1a603b177733c) for this 🙏

[the remark this answers is not in the document]
Comment (louischan-oursky) [reply e2e/tests/login/totp/totp_user.sql]: No, we do not support importing recovery codes.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**